htdocs/salaries/card.php

```diff
@@ -738,6 +738,22 @@ function( data ) {
 		$formquestion[] = array('type' => 'text', 'name' => 'amount', 'label' => $langs->trans("Amount"), 'value' => price($object->amount), 'morecss' => 'width100 right');
 
 		$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"].'?id='.$object->id, $langs->trans('ToClone'), $langs->trans('ConfirmCloneSalary', $object->ref), 'confirm_clone', $formquestion, 'yes', 1, 280);
+
+		//Add fill with end of month button
+		$formconfirm .= "<script>
+			$('#clone_date_end').after($('<button id=\"fill_end_of_month\" class=\"dpInvisibleButtons\" style=\"color: var(--colortextlink);font-size: 0.8em;opacity: 0.7;margin-left:4px;\" type=\"button\">".$langs->trans('FillEndOfMonth')."</button>'));
+			$('#fill_end_of_month').click(function(){
+				var jsDate = $('#clone_date_start').datepicker('getDate');
+				if (jsDate !== null) {
+					jsDate instanceof Date;
+					var lastDate = new Date(jsDate.getFullYear(), jsDate.getMonth() + 1, 0);
+					$('#clone_date_endday').val(lastDate.getDate());
```

Comment:
You get the date using 
var jsDate = $('#clone_date_start').datepicker('getDate');  then  lastDate.getDate() for the day, getMonth for the month, ...
But you should be able to get the value with 
$('#clone_date_startday').val()
$('#clone_date_startmonth').val()
...
Because the day, month and year are already always saved as structure data as hidden field when we have a date. 
The advantage is that your new features does not depends on datepicker. If we replace datepicker with another js component to fill date_start and its strcutured data, the code will continue to work.

Reply:
@eldy 
I added the button without using the datepicker() library. 
I used the variable $langs->shortlang, which returns en_US or fr_FR .... 
I didn't find any variable that returns en-US instead of en_US.